Bot/modules/db.module.js

```diff
@@ -0,0 +1,272 @@
+'use strict';
+
+const UIDGenerator = require('uid-generator');
+const pool = require('../../DB/pool');
+const config = require('../../config');
+
+process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = 0;
+process.on('unhandledRejection', error => {
+  console.log(error);
+  process.exit();
+});
+process.on('exit', () => {
+  pool.end();
+});
+
+const uidgen = new UIDGenerator();
+
+function insertData(query, ctx, message) { // for insert/create query
+  const promise = new Promise(async (resolve, reject) => {
+    const client = await pool.connect();
+    try {
+      await client.query(query);
+      await client.release();
+      if (ctx) ctx.reply(message);
+      resolve();
+    } catch (e) {
+      await client.release();
+      console.log(e);
+      if (ctx) ctx.reply('Error');
+      reject();
+    }
+  });
+  return promise;
+}
+
+function selectData(query, responseHandler = x => x) { // for select query
+  const promise = new Promise(async (resolve, reject) => {
+    const client = await pool.connect();
+    try {
+      const response = await client.query(query);
+      await client.release();
+      resolve(responseHandler(response));
+    } catch (e) {
+      console.log(e);
+      await client.release();
+      reject();
+    }
+  });
+  return promise;
+}
+
+function insertCourse(ctx, courses, userID, username) { // adds course for teacher / register teacher if needed
+  const currCourse = courses[userID];
+  const query = `INSERT INTO Courses VALUES
+  ('${currCourse.courseID}', '${currCourse.courseName}', ${userID}, 
+    ${currCourse.labNumb}, ${currCourse.testNumb}, ${+currCourse.additional})`;
+  const promise = new Promise(async (resolve, reject) => {
+    const client = await pool.connect();
+    try {
+      await client.query(query);
+      await client.release();
+      ctx.reply(config.messages.successCourseMessage);
+      const isRegistred = await isTeacherRegistred(userID);
+      if (!isRegistred) {
+        await insertTeacher(ctx, userID, username);
+      }
+      resolve();
+    } catch (e) {
+      await client.release();
+      console.log(e);
+      ctx.reply(config.messages.dismissMessage);
+      reject();
+    }
+  });
+  return promise;
+}
+
+async function deleteCourse(courseName, userID) {
+  const template = `FROM public.courses
+	WHERE course_name='${courseName}' AND teacher_id=${userID}`;
+  const getCourseID = 'SELECT course_id ' + template;
+  const courseID = await selectData(getCourseID, res => res.rows[0].course_id);
+  const query1 = 'DELETE ' + template;
+  const query2 = `DELETE FROM public.courses_groups
+	WHERE course_id='${courseID}'`;
```

Comment:
Warning: place for SQL injections!
Don't create SQL with string concatenation with parameters, pass parameters separately as a collection.
![ee7bcc72b339cb91910fb4fd589ccb8e](https://user-images.githubusercontent.com/4405297/120098858-ee98ef80-c140-11eb-9896-f34ed9c4505a.gif)